### PR TITLE
Rig Calibration with different stream lenghts

### PR DESCRIPTION
**Pull request by simogasp**
Allow to calibrate a rig with streams of different lengths (yet they have to be synchronized)

### CODE REVIEW

**Comment by simogasp** [comment]
Need update and some editing for the last commit about logging

File: src/openMVG/rig/Rig.cpp [hunk 1]
```diff
@@ -67,17 +67,16 @@ bool Rig::initializeCalibration()
...
-  // check that all the cameras have the same number of localizationResults
-  const std::size_t sequenceLength = _vLocalizationResults[0].size();
+  // make all the cameras have the same number of localizationResults (equal to the shortest clip)
   const std::size_t nCams = _vLocalizationResults.size();
-  
-  for(std::size_t i = 1; i < nCams; ++i)
   {
-    assert(_vLocalizationResults[i].size() == sequenceLength 
-            && "All cameras must have the same number of images");
+    std::size_t shortestSeqLength(-1);
```
Comment (simogasp) [inline src/openMVG/rig/Rig.cpp]: @zvrba wouldn't be better
```c++
std::size_t shortestSeqLength = std::numeric_limits<std::size_t>::max();
```
I mean, in the end the effect should be the same but we avoid the overflow

**Comment by simogasp** [comment]
i force-pushed without the last commit as it is fixed in develop, we'll have a more clear history.

**Comment by simogasp** [comment]
Actually this branch has already be merged at some point in the past. The only part that is not in the develop branch is this one

https://github.com/poparteu/openMVG/pull/233/commits/73b5c51c8f72a1a5d2e2f8f2b1beb401a4667783#diff-3a4de51d91f6e4fd1361f5d98f8f7c9bR395

which allow to set/unset use of CUDA for CCtag.

So I propose to close this PR without merging and we'll see how to manage the CUDA at runtime.

@fabiencastan @haavares @zvrba

**Comment by haavares** [comment]
ok

File: src/openMVG/rig/Rig.cpp (hunk 1 above)
Comment (zvrba) [inline src/openMVG/rig/Rig.cpp]: There's no overflow, `size_t(-1)` is a well-defined way of creating a `size_t` of maximum possible value. (On 2nd complement machines, but I've yet to see one which is based on one's complement AND runs openmvg).

**Pull request closed without merging**